### PR TITLE
[Transaction] Move MetaStoreHandlerNotReadyException and add MetaStoreHandlerHasClosedException

### PR DESCRIPTION
**Pull request by liangyepianzhou**
Master https://github.com/apache/pulsar/issues/13918

### Motivation & Modification
Makes exceptions on the Transaction client side clearer.

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly (if you have committer privilege).

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Comment by gaoran10** [comment]
Sorry, I'm not sure why we need to make the exception `MetaStoreHandlerNotReadyException` to extend `PulsarClientException`. Could you add some explanation?

File: pulsar-broker/src/test/java/org/apache/pulsar/client/impl/TransactionClientConnectTest.java [hunk 1]
```diff
@@ -156,8 +157,7 @@ public void testPulsarClientCloseThenCloseTcClient() throws Exception {
             try {
                 handler.newTransactionAsync(10, TimeUnit.SECONDS).get();
```
Comment (mattisonchao) [inline pulsar-broker/src/test/java/org/apache/pulsar/client/impl/TransactionClientConnectTest.java:158]: Maybe we need to add ``fail()`` here.

**Comment by tisonkun** [comment]
Closed as stale and conflict. Please rebase and resubmit the patch if it's still relevant.

**Pull request closed without merging**